### ISSUE TITLE
Unable to pass event NodeAddEvent to handler github.scarsz.discordsrv.hooks.permissions.LuckPermsHook

### ISSUE THREAD

**Issue by mibby** [lucko/LuckPerms#2120]
LuckPerms dev 1025

```
[06:47:00] [ForkJoinPool-2-worker-37/WARN]: [LuckPerms] Unable to pass event NodeAddEvent to handler github.scarsz.discordsrv.hooks.permissions.LuckPermsHook$$Lambda$5036/0x00007fd5467950b0
[06:47:00] [ForkJoinPool-2-worker-37/WARN]: java.lang.ExceptionInInitializerError
[06:47:00] [ForkJoinPool-2-worker-37/WARN]: 	at me.lucko.luckperms.common.util.PrivateMethodHandles.<clinit>(PrivateMethodHandles.java:41)
[06:47:00] [ForkJoinPool-2-worker-37/WARN]: 	at me.lucko.luckperms.common.event.gen.GeneratedEventSpec$EventInvocationHandler.invoke(GeneratedEventSpec.java:152)
[06:47:00] [ForkJoinPool-2-worker-37/WARN]: 	at com.sun.proxy.$Proxy126.isUser(Unknown Source)
[06:47:00] [ForkJoinPool-2-worker-37/WARN]: 	at github.scarsz.discordsrv.hooks.permissions.LuckPermsHook.handle(LuckPermsHook.java:38)
[06:47:00] [ForkJoinPool-2-worker-37/WARN]: 	at github.scarsz.discordsrv.hooks.permissions.LuckPermsHook.lambda$new$1(LuckPermsHook.java:33)
[06:47:00] [ForkJoinPool-2-worker-37/WARN]: 	at me.lucko.luckperms.common.event.LuckPermsEventSubscription.invoke(LuckPermsEventSubscription.java:95)
[06:47:00] [ForkJoinPool-2-worker-37/WARN]: 	at me.lucko.luckperms.common.event.LuckPermsEventSubscription.invoke(LuckPermsEventSubscription.java:43)
[06:47:00] [ForkJoinPool-2-worker-37/WARN]: 	at me.lucko.luckperms.lib.eventbus.SimpleEventBus.post(SimpleEventBus.java:107)
[06:47:00] [ForkJoinPool-2-worker-37/WARN]: 	at me.lucko.luckperms.common.event.AbstractEventBus.post(AbstractEventBus.java:84)
[06:47:00] [ForkJoinPool-2-worker-37/WARN]: 	at me.lucko.luckperms.common.event.EventDispatcher.post(EventDispatcher.java:116)
[06:47:00] [ForkJoinPool-2-worker-37/WARN]: 	at me.lucko.luckperms.common.event.EventDispatcher.lambda$post$0(EventDispatcher.java:129)
[06:47:00] [ForkJoinPool-2-worker-37/WARN]: 	at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1429)
[06:47:00] [ForkJoinPool-2-worker-37/WARN]: 	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
[06:47:00] [ForkJoinPool-2-worker-37/WARN]: 	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1016)
[06:47:00] [ForkJoinPool-2-worker-37/WARN]: 	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1665)
[06:47:00] [ForkJoinPool-2-worker-37/WARN]: 	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1598)
[06:47:00] [ForkJoinPool-2-worker-37/WARN]: 	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:177)
[06:47:00] [ForkJoinPool-2-worker-37/WARN]: Caused by: java.lang.NoSuchMethodException: java.lang.invoke.MethodHandles$Lookup.<init>(java.lang.Class,int)
[06:47:00] [ForkJoinPool-2-worker-37/WARN]: 	at java.base/java.lang.Class.getConstructor0(Class.java:3427)
[06:47:00] [ForkJoinPool-2-worker-37/WARN]: 	at java.base/java.lang.Class.getDeclaredConstructor(Class.java:2631)
[06:47:00] [ForkJoinPool-2-worker-37/WARN]: 	at me.lucko.luckperms.common.util.PrivateMethodHandles.<clinit>(PrivateMethodHandles.java:38)
[06:47:00] [ForkJoinPool-2-worker-37/WARN]: 	... 16 more
[06:47:02] [ForkJoinPool-2-worker-37/WARN]: [LuckPerms] Unable to pass event NodeAddEvent to handler github.scarsz.discordsrv.hooks.permissions.LuckPermsHook$$Lambda$5036/0x00007fd5467950b0
[06:47:02] [ForkJoinPool-2-worker-37/WARN]: java.lang.NoClassDefFoundError: Could not initialize class me.lucko.luckperms.common.util.PrivateMethodHandles
[06:47:02] [ForkJoinPool-2-worker-37/WARN]: 	at me.lucko.luckperms.common.event.gen.GeneratedEventSpec$EventInvocationHandler.invoke(GeneratedEventSpec.java:152)
[06:47:02] [ForkJoinPool-2-worker-37/WARN]: 	at com.sun.proxy.$Proxy126.isUser(Unknown Source)
[06:47:02] [ForkJoinPool-2-worker-37/WARN]: 	at github.scarsz.discordsrv.hooks.permissions.LuckPermsHook.handle(LuckPermsHook.java:38)
[06:47:02] [ForkJoinPool-2-worker-37/WARN]: 	at github.scarsz.discordsrv.hooks.permissions.LuckPermsHook.lambda$new$1(LuckPermsHook.java:33)
[06:47:02] [ForkJoinPool-2-worker-37/WARN]: 	at me.lucko.luckperms.common.event.LuckPermsEventSubscription.invoke(LuckPermsEventSubscription.java:95)
[06:47:02] [ForkJoinPool-2-worker-37/WARN]: 	at me.lucko.luckperms.common.event.LuckPermsEventSubscription.invoke(LuckPermsEventSubscription.java:43)
[06:47:02] [ForkJoinPool-2-worker-37/WARN]: 	at me.lucko.luckperms.lib.eventbus.SimpleEventBus.post(SimpleEventBus.java:107)
[06:47:02] [ForkJoinPool-2-worker-37/WARN]: 	at me.lucko.luckperms.common.event.AbstractEventBus.post(AbstractEventBus.java:84)
[06:47:02] [ForkJoinPool-2-worker-37/WARN]: 	at me.lucko.luckperms.common.event.EventDispatcher.post(EventDispatcher.java:116)
[06:47:02] [ForkJoinPool-2-worker-37/WARN]: 	at me.lucko.luckperms.common.event.EventDispatcher.lambda$post$0(EventDispatcher.java:129)
[06:47:02] [ForkJoinPool-2-worker-37/WARN]: 	at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1429)
[06:47:02] [ForkJoinPool-2-worker-37/WARN]: 	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
[06:47:02] [ForkJoinPool-2-worker-37/WARN]: 	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1016)
[06:47:02] [ForkJoinPool-2-worker-37/WARN]: 	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1665)
[06:47:02] [ForkJoinPool-2-worker-37/WARN]: 	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1598)
[06:47:02] [ForkJoinPool-2-worker-37/WARN]: 	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:177)
```

**Comment by lucko** [comment]
Java version?

**Comment by mibby** [comment]
Java 14.

**Comment by lucko** [comment]
Should be fixed by c3128dec0d3cb955ffbe8b8bed7eae66d4b7fd33 - however, this commit introduces a completely different approach to events which may itself have bugs.

I've done some initial testing and it all seems to work ok though!

Thank you for reporting.

**Comment by mibby** [comment]
Thanks for the quick fix! I'll let you know if I come across any further issues with the new approach.
